llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -883,6 +883,8 @@ class VPInstruction : public VPRecipeWithIRFlags,
     AnyOf,
     // Calculates the first active lane index of the vector predicate operand.
     FirstActiveLane,
+    // Creates a step vector starting from 0 with a step of 1.
```

Comment:
```suggestion
    // Creates a step vector starting from 0 with a step of 1 to the end value specified in the first operand.
```

It would probably be better to specify the end value/VF as operand?

Reply:
I don't think the stepvector intrinsic has the ability to specify an end value?

Reply:
Or Builder.CreateStepVector doesn't have an API for it either, i.e. it always creates a step vector for the full vector width, for both scalable and fixed-length vectors. And we don't have a VP intrinsic for stepvector i.e. https://github.com/llvm/llvm-project/issues/121561

Reply:
ah right, it needs to take the type. Would be convenient to have it as operand for some transforms (e.g. `narrowInterleaveGroups` but ok to leave it as-is for now.. Still good to document the range